sorting_algorithms/cocktail_shaker_sort.pl

```diff
@@ -0,0 +1,55 @@
+% swap_up(+List, -ListWithMovedElement)
+% goes through the list
+% checks every two adjacent numbers
+% if the first is greater, then swap them
+% in other words, the biggest element is moved to position where after its there are only greater numbers.
+swap_up([], []).
+swap_up([X], [X]).
+swap_up([Fst, Snd|T],[Snd|L]) :- Fst > Snd, swap_up([Fst|T], L), !.
+swap_up([Fst, Snd|T],[Fst|L]) :- Fst =< Snd, swap_up([Snd|T], L).
+
+test_swap_up(X) :- swap_up([4,3,2,1], X).
+% X = [3,2,1,4]
+
+
+% swap_down(+List, -ListWithMovedElement)
+% goes through the list
+% checks every two adjacent numbers
+% if the first is smaller, then swap them
+% in other words, the smallest element is moved to position where after its there are only smaller numbers.
+swap_down([], []).
+swap_down([X], [X]).
+swap_down([Fst, Snd|T],[Snd|L]) :- Fst < Snd, swap_down([Fst|T], L), !.
+swap_down([Fst, Snd|T],[Fst|L]) :- Fst >= Snd, swap_down([Snd|T], L).
```

Comment:
```suggestion
swap_down([Fst, Snd|T],[Fst|L]) :- swap_down([Snd|T], L).
```
Sorry, zapomniałem przy ostatnim:
jak masz odcięcie w poprzedniej lini to tu nie musisz sprawdzać Fst =< Snd